bin/pycbc_convertinjfiletohdf

```diff
@@ -0,0 +1,194 @@
+
+import pycbc
+from pycbc.inject import InjectionSet, CBCHDFInjectionSet
+from pycbc.types import float64, float32
+from pycbc.io.record import FieldArray
+
+from pycbc.io.ligolw import LIGOLWContentHandler
+from ligo.lw import utils as ligolw_utils, lsctables
+
+
+# Handlers for alternative injection formats
+
+def legacy_approximant_name(apx):
+    """Convert the old style xml approximant name to a name
+    and phase_order.
+    """
+    import lalsimulation as lalsim
+    apx = str(apx)
+    try:
+        order = lalsim.GetOrderFromString(apx)
+    except:
+        print("Warning: Could not read phase order from string, using default")
+        order = -1
+    name = lalsim.GetStringFromApproximant(lalsim.GetApproximantFromString(apx))
+    return name, order
+
+
+class XMLInjectionSet(object):
+
+    """Reads injections from LIGOLW XML files
+
+    Parameters
+    ----------
+    sim_file : string
+        Path to a LIGOLW XML file containing a SimInspiralTable
+        with injection definitions.
+
+    Attributes
+    ----------
+    indoc
+    table
+    """
+
+    def __init__(self, sim_file):  
+        self.indoc = ligolw_utils.load_filename(
+            sim_file, False, contenthandler=LIGOLWContentHandler)
+        self.table = lsctables.SimInspiralTable.get_table(self.indoc)
+
+    def end_times(self):
+        """Return the end times of all injections"""
+        return [inj.time_geocent for inj in self.table]
+
+    def pack_data_into_pycbc_format_input(self):
+        data = {}
+        for key in self.table[0].__slots__:
+            # Some XML files can have empty columns which are read as None in
+            # python. For these cases we ignore the columns so they do not
+            # appear in the output HDF file. (Such XML files cannot currently
+            # be read by LALSuite C code.) 
+            if getattr(self.table[0], key) is not None:
+                data[str(key)] = numpy.array(
+                    [getattr(t, key) for t in self.table]
+                )
+
+        for k in ['simulation_id', 'process_id']:
+            a = data.pop(k)
+
+        data['approximant'], data['phase_order'] = numpy.array(
+                [legacy_approximant_name(wf) for wf in data['waveform']]
+        ).T
+        data['tc'] = (
+            data['geocent_end_time'] + 1e-9 * data['geocent_end_time_ns']
+        )
+        data['dec'] = data['latitude']
+        data['ra'] = data['longitude']
+        return data
+
+
+class LVKNewStyleInjectionSet(object):
+    """Reads injections from new-style LVK injection files
+    """
+
+    translation_dict = {
+        'mass1_det': 'mass1',
+        'mass2_det': 'mass2',
+        'f22_ref_spin': 'f_ref',
+        't_co_gps': 'tc',
+        'd_lum': 'distance',
+        'f22_start': 'f_lower',
+        'cbc_model': 'approximant',
+        'longAscNodes': 'long_asc_nodes',
+        'eccentricity': 'eccentricity',
+        'meanPerAno': 'mean_per_ano',
+        't_co_gps_add': None,
```

Comment:
I want it to be clear in the code that we are handling this column. So even though we could remove it , I do want to explicitly include it here. Also, if anyone does change the code at a later date, the code will then fail if trying to add this column, rather than copying a t_co_gps_add entry across, which would be ignored.